src/catalog/catalog_accessor.cpp

```diff
@@ -17,7 +17,23 @@ db_oid_t CatalogAccessor::CreateDatabase(std::string name) {
 
 bool CatalogAccessor::DropDatabase(db_oid_t db) { return catalog_->DeleteDatabase(txn_, db); }
 
-void CatalogAccessor::SetSearchPath(std::vector<namespace_oid_t> namespaces) { search_path_ = std::move(namespaces); }
+void CatalogAccessor::SetSearchPath(std::vector<namespace_oid_t> namespaces) {
+  TERRIER_ASSERT(!namespaces.empty(), "search path cannot be empty");
+  default_namespace_ = namespaces[0];
+
+  for (auto ns : namespaces) {
+    // Check if 'pg_catalog is explicitly set'
+    if (ns == NAMESPACE_CATALOG_NAMESPACE_OID) {
+      search_path_ = std::move(namespaces);
+      return;
+    }
+  }
+
+  search_path_.clear();
+  search_path_.reserve(namespaces.size() + 1);
+  search_path_.emplace_back(NAMESPACE_CATALOG_NAMESPACE_OID);
```

Comment:
You'd get equivalent behavior from...
```
search_path_ = std::move(namespaces);
search_path_.emplace(search_path_ .begin(), NAMESPACE_CATALOG_NAMESPACE_OID);
```
... right? Complexity is the same, but the behavior is more clear to me. Might just be me though, not an explicit change request.